code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +40,184 @@ internal sealed class AEPPushTemplate {
-        body = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.BODY_TEXT)
-            ?: throw IllegalArgumentException("Required field \"${PushTemplateIntentConstants.IntentKeys.BODY_TEXT}\" not found.")
-        payloadVersion = intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.PAYLOAD_VERSION)
-        payloadVersion?.let {
-            if (it < 1) throw IllegalArgumentException("Invalid \"${PushTemplateIntentConstants.IntentKeys.PAYLOAD_VERSION}\" found.")
-        }
-
-        // optional values
-        sound = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.CUSTOM_SOUND)
-        imageUrl = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.IMAGE_URI)
-        actionUri = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.ACTION_URI)
-        actionType = PushTemplateConstants.ActionType.valueOf(
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.ACTION_TYPE)
-                ?: PushTemplateConstants.ActionType.NONE.name
-        )
-        smallIcon = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.SMALL_ICON)
-        largeIcon = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.LARGE_ICON)
-        badgeCount = intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.BADGE_COUNT)
-        notificationPriority = intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.PRIORITY)
-        notificationImportance =
-            intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.IMPORTANCE)
-        notificationVisibility =
-            intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.VISIBILITY)
-        channelId = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.CHANNEL_ID)
-        templateType =
-            PushTemplateType.fromString(intentExtras.getString(PushTemplateIntentConstants.IntentKeys.TEMPLATE_TYPE))
-        tag = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.TAG)
-        isNotificationSticky =
-            intentExtras.getBoolean(PushTemplateIntentConstants.IntentKeys.STICKY)
-        ticker = intentExtras.getString(PushTemplateIntentConstants.IntentKeys.TICKER)
-        expandedBodyText =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.EXPANDED_BODY_TEXT)
-        expandedBodyTextColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.EXPANDED_BODY_TEXT_COLOR)
-        titleTextColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.TITLE_TEXT_COLOR)
-        smallIconColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.SMALL_ICON_COLOR)
-        notificationBackgroundColor =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.NOTIFICATION_BACKGROUND_COLOR)
-        isFromIntent = true
+    internal open fun createIntent(action: String): Intent {
```

Comment:
Can we move this to either `AEPPushNotificationBuilder` or any of the extension classes? The template classes only have the data we need for the notificatio construction

Reply:
I missed this. Good point! I am also inclined to move it to the builder.